homeassistant/components/sensor/__init__.py

```diff
@@ -305,6 +307,23 @@ def state(self) -> Any:
                 temp = units.temperature(float(value), unit_of_measurement)
                 value = round(temp) if prec == 0 else round(temp, prec)
 
+        if value is not None and device_class in (
+            DEVICE_CLASS_TIMESTAMP,
+            DEVICE_CLASS_DATE,
+        ):
+            try:
+                ciso8601.parse_datetime(str(value))
+            except (ValueError, IndexError):
```

Comment:
We don’t currently validate any data on runtime but instead we treat it as a bug in the integration and fix that. I did once start validator integration to find these so bug reporting is easier.

Reply:
This is no longer correct, we validate a lot now.